### PR TITLE
[jsk_perception] Keep original resolution if all the input images has same shape and add ~draw_input_topic parameter to draw topic name on the tiled images

### PR DESCRIPTION
**Pull request by garaemon**
screenshot_from_2015-12-20 18 11 30

Modified:
  - jsk_perception/node_scripts/tile_image.py
  - jsk_recognition_utils/python/jsk_recognition_utils/visualize.py

### CODE REVIEW

**Comment by garaemon** [comment]
review @wkentaro

File: jsk_perception/node_scripts/tile_image.py [hunk 1]
```diff
@@ -47,8 +48,10 @@ def unsubscribe(self):
     def _apply(self, *msgs):
         bridge = cv_bridge.CvBridge()
         imgs = []
-        for msg in msgs:
+        for msg, topic in zip(msgs, self.input_topics):
             img = bridge.imgmsg_to_cv2(msg, desired_encoding='bgr8')
+            if self.draw_topic_name:
```
Comment (wkentaro) [inline jsk_perception/node_scripts/tile_image.py]: Maybe you don't need to set value to property but directly see the rosparam.
like
```python
if rospy.get_param('~draw_topic_name', False):
   ...
```

File: jsk_recognition_utils/python/jsk_recognition_utils/visualize.py [hunk 2]
```diff
@@ -28,18 +27,40 @@ def get_tile_shape(img_num):
...
-    canvas = plt.get_current_fig_manager().canvas
-    canvas.draw()
-    pil_img = PIL.Image.frombytes('RGB',
-        canvas.get_width_height(), canvas.tostring_rgb())
-    out_rgb = np.array(pil_img)
-    out_bgr = cv2.cvtColor(out_rgb, cv2.COLOR_RGB2BGR)
-    plt.close()
-    return out_bgr
+    if all([(img.shape[0] == imgs[0].shape[0] and
+             img.shape[1] == imgs[0].shape[1])
```
Comment (wkentaro) [inline jsk_recognition_utils/python/jsk_recognition_utils/visualize.py]: You can write as 
```python
if all(img.shape == imgs[0].shape for img in imgs):
  ....
```

File: jsk_recognition_utils/python/jsk_recognition_utils/visualize.py [hunk 3]
```diff
@@ -28,18 +27,40 @@ def get_tile_shape(img_num):
...
+        concatenated_image = None
+        for y in range(y_num):
+            row_image = None
+            for x in range(x_num):
+                i = x + y * x_num
+                if i >= len(imgs):
+                    img = np.zeros(imgs[0].shape, dtype=np.uint8)
+                else:
+                    img = imgs[i]
+                if row_image == None:
```
Comment (wkentaro) [inline jsk_recognition_utils/python/jsk_recognition_utils/visualize.py]: it should be
```python
if row_image is None:
   ...
```

File: jsk_recognition_utils/python/jsk_recognition_utils/visualize.py [hunk 4]
```diff
@@ -28,18 +27,40 @@ def get_tile_shape(img_num):
...
+                    row_image = img
+                else:
+                    row_image = cv2.hconcat([row_image, img])
+            if concatenated_image == None:
```
Comment (wkentaro) [inline jsk_recognition_utils/python/jsk_recognition_utils/visualize.py]: same as above

**Comment by wkentaro** [comment]
Other than above comments, looks good to me. It's nice feature. :sparkles:

File: jsk_perception/node_scripts/tile_image.py (hunk 1 above)
Comment (garaemon) [inline jsk_perception/node_scripts/tile_image.py]: get_param invokes communication to roscore and I'd like to remove the overhead

Comment (wkentaro) [inline jsk_perception/node_scripts/tile_image.py]: I see.

**Pull request merged**